contractsV5/KyberNetwork.sol

```diff
@@ -678,7 +678,56 @@ contract KyberNetwork is Withdrawable, Utils, IKyberNetwork, ReentrancyGuard {
     function handleFees(TradeData memory tradeData) internal returns(bool) {
         // create array of reserves receiving fees + fee percent per reserve
         // fees should add up to 100%.
-        // send total fee amount to fee handler with reserve data.
+
+        uint length = tradeData.numFeePayingReserves;
+        address[] memory eligibleReserves = new address[](length);
+        uint[] memory rebatePercentages = new uint[](length);
+
+        // Updates reserve eligibility and rebate percentages
+        updateEligibilityAndRebates(eligibleReserves, rebatePercentages, tradeData);
+
+        // Sending platform fee to taker platform
+        uint platformFeeWei = tradeData.platformFeeWei;
+        tradeData.platformFeeWei = 0; // To protect against reentrancy attacks
+        (bool success, ) = tradeData.input.platformWallet.call.value(platformFeeWei)("");
+        require(success, "Transfer failed");
```

Comment:
in general its better to have optional reverting code sooner then later.
to save user gas.
so this could be done sooner.

Reply:
noted.